sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -323,18 +323,25 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
           provider match {
             case supportsExtract: SupportsCatalogOptions =>
               val ident = supportsExtract.extractIdentifier(dsOptions)
-              val catalog = CatalogV2Util.getTableProviderCatalog(
-                supportsExtract, catalogManager, dsOptions)
+              val catalog = if (ident.namespace().isEmpty) {
+                Array(dsOptions.get("catalog"))
+              } else {
+                ident.namespace()
+              }
 
               val location = Option(dsOptions.get("path")).map(TableCatalog.PROP_LOCATION -> _)
-
+              val tableSpec = TableSpec(None, Map.empty, Some(source), Map.empty,
```

Comment:
there are a lot of literals here, let's call the method using named parameters, e.g. `TableSpec(bucketSpec = None, ...)`